pywbem_mock/_defaultinstanceprovider.py

```diff
@@ -0,0 +1,527 @@
+            raise CIMError(
+                CIM_ERR_INVALID_CLASS,
+                _format("Class {0!A} for GetInstance of instance {1!A} "
+                        "does not exist.", iname.classname, iname))
+
+        return self._get_instance(iname, instance_store,
+                                  LocalOnly,
+                                  IncludeClassOrigin,
+                                  IncludeQualifiers, PropertyList)
+
+    ####################################################################
+    #
+    #   DeleteInstance server request
+    #
+    ####################################################################
+
+    def DeleteInstance(self, InstanceName):
+        """
+        Implements a WBEM server responder for
+        :meth:`pywbem.WBEMConnection.DeleteInstance`.
+
+        This method deletes a single instance from the CIM repository based on
+        the  InstanceName parameter.
+
+        It does not attempt to delete referencing instances (associations,
+        etc.) that reference this instance.
+
+        If the creation class of the instance to be deleted is PG_Namespace or
+        CIM_Namespace, then the namespace identified by the 'Name' property
+        of the instance is being deleted in the CIM repository in addition to
+        the CIM instance.
+
+        NOTE: This method includes namespace within the InstanceName
+        rather than as a separate input argument
+
+        Parameters:
+
+          InstanceName (:class:`~pywbem.CIMInstanceName`):
+            The instance path of the instance to be deleted.
+            The instance path of the instance to be retrieved  with the
+            following attributes:
+
+            * `classname`: Name of the creation class of the instance.
+            * `keybindings`: Keybindings of the instance.
+            * `namespace`: Name of the CIM namespace containing the instance.
+              Must not be None.
+            * `host`: value ignored.
+
+        Raises:
+            CIMError: CIM_ERR_INVALID_NAMESPACE
+            CIMError: CIM_ERR_INVALID_CLASS
+            CIMError: CIM_ERR_NOT_FOUND
+        """
+
+        namespace = InstanceName.namespace
+        self.validate_namespace(namespace)
+
+        iname = InstanceName
+
+        # Validate namespace and get instance CIM repository
+        instance_store = self.get_instance_store(namespace)
+
+        if not self._class_exists(namespace, iname.classname):
+            raise CIMError(
+                CIM_ERR_INVALID_CLASS,
+                _format("Class {0!A} in namespace {1!A} not found. "
+                        "Cannot delete instance {2!A}",
+                        iname.classname, namespace, iname))
+
+        if not instance_store.exists(iname):
+            raise CIMError(
+                CIM_ERR_NOT_FOUND,
+                _format("Instance {0!A} not found in CIM repository namespace "
+                        "{1!A}", iname, namespace))
```

Comment:
This check could be done before calling the provider.

Reply:
It is now done in the ProviderDispatcher method.  DONE